### PR TITLE
chore: Meltano `v3`

### LINKED ISSUE #7957: docs: Meltano V2 -> V3 migration guide
This should note all removals, deprecations, feature flags made default, dependencies that now require an extra, and other significant changes that a user may need to act on in order to upgrade from Meltano V2 to V3. This is separate from the changelog.

Consider the Meltano V1 -> V2 migration guide as an example: https://docs.meltano.com/guide/v2-migration/

We may want to make a section in the docs for version upgrade migration guides, rather than listing them all directly under the guides section. If we do this, we'll have to add a redirect from the existing V1 -> V2 migration guide to its new location.

### LINKED ISSUE #7958: docs: Meltano V3 release notes
As per usual, the changelog for Meltano `v3.0.0` will be automatically generated based off of commits, and the release notes will start off as a draft of what's in the changelog. For such a significant release, we'll want to do more than the usual clean-up we do to the changelog to produce the release notes.

We should call attention to new features, and to [the V2 -> V3 migration guide](https://github.com/meltano/meltano/issues/7957).

This should be done after the `cloud` branch has been merged into `main` (by rebase rather than squash). Once that has been done, we can create a draft release as a starting point for these release notes.
- https://github.com/meltano/meltano/pull/7869

### LINKED ISSUE #8037: Create `v3` branch from `cloud` branch without the Cloud CLI
@meltano/engineering I had a brief chat with @WillDaSilva about cutting a v3 release to help with velocity so v2 doesn't have to keep being maintained along side the `cloud` branch. 

Can we pull out the cloud specific features from the `cloud` branch and only merge the updates to Core from there?

### PR DESCRIPTION
**Pull request by WillDaSilva**
Closes #8037

Closes #7958

Closes #7957

Supersedes:
- https://github.com/meltano/meltano/pull/7869

Related milestone:
- https://github.com/meltano/meltano/milestone/13

Similar to the above PR, we'd only want to merge this PR into `main` with a rebase merge. We'll have to temporarily change the repository settings to allow that, if/when this PR has been approved and is ready to merge.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -1,76 +1,122 @@
...
+                "COVERAGE_FILE": str(
+                    root_path / f".coverage.{random_seed:010}.{session.name}",
+                ),
+            },
+        )
+        session.run(
+            "pytest",
+            "--cov=meltano",
+            "--cov=tests",
+            "tests/",
```
Comment (edgarrmondragon) [inline noxfile.py:78]:
```suggestion
```

File: noxfile.py [hunk 2]
```diff
@@ -1,76 +1,122 @@
...
+    "pytest-randomly",
+    "pytest-xdist",
+    "requests-mock",
+    "pytest-structlog",
+)
+
+
+def _run_pytest(session: Session) -> None:
+    random_seed = randint(0, 2**32 - 1)  # noqa: S311, WPS432
+    try:
```
Comment (edgarrmondragon) [inline noxfile.py:65]:
```suggestion
    args = session.posargs or ("tests/",)
    try:
```

File: noxfile.py [hunk 3]
```diff
@@ -1,76 +1,122 @@
...
+                ),
+            },
+        )
+        session.run(
+            "pytest",
+            "--cov=meltano",
+            "--cov=tests",
+            "tests/",
+            f"--randomly-seed={random_seed}",
+            *session.posargs,
```
Comment (edgarrmondragon) [inline noxfile.py:80]:
```suggestion
            *args,
```

File: docs/docs/guide/v3-migration.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+title: Meltano 3.0 Migration Guide
+description: Migrate existing "v2" projects to the latest version of Meltano
+layout: doc
+sidebar_position: 21
+sidebar_class_name: hidden
```
Comment (tayloramurphy) [inline docs/docs/guide/v3-migration.md:6]:
```suggestion
```

**Review by edgarrmondragon — APPROVED** [review]

**Comment by tayloramurphy** [comment]
@WillDaSilva it looks like the pages under `/cloud` still have the change to `meltano cloud`. Other than that I think this looks good.

**Comment by WillDaSilva** [comment]
> it looks like the pages under `/cloud` still have the change to `meltano cloud`

@tayloramurphy Addressed in 46fc4835c082d106d37ce28a1f5f37ec377aa5c1

**Review by tayloramurphy — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,30 @@
...
+
+### 🐛 Fixes
+
+- [#8042](https://github.com/meltano/meltano/issues/8042) Ensure migration lock file is closed after read
+- [#8043](https://github.com/meltano/meltano/issues/8043) Ensure elt/run log file is closed
+
+### ⚙️ Under the Hood
+
+- [#7960](https://github.com/meltano/meltano/issues/7960) Compatibility with SQLAlchemy 2
+
```
Comment (edgarrmondragon) [inline CHANGELOG.md:29]: This should include the changes from https://github.com/meltano/meltano/pull/8036/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed

Comment (WillDaSilva) [reply CHANGELOG.md:39]: Good catch. Included in 07a5945

**Comment by edgarrmondragon** [comment]
Are we good to merge? `3.0.0rc0` is in PyPI 🙂

**Comment by WillDaSilva** [comment]
To do after merging this:
- Disable rebase merges into `main` for this repository

**Comment by edgarrmondragon** [comment]
@WillDaSilva so you think I'm clear to hit rebase and merge?

**Comment by WillDaSilva** [comment]
@edgarrmondragon I believe we're good to go

**Comment by WillDaSilva** [comment]
@edgarrmondragon Thoughts on using a regular merge commit then? Instead of a squash merge, that is

**Comment by edgarrmondragon** [comment]
> @edgarrmondragon Thoughts on using a regular merge commit then? Instead of a squash merge, that is

@WillDaSilva yeah I think that's the next best thing. There's no way we can resolve all the conflicts in 150+ commits. I'll go ahead and enable merge commits.

**Pull request merged**